src/connect.rs

```diff
@@ -444,12 +543,12 @@ impl Service<Uri> for Connector {
     }
 }
 
-pub(crate) trait AsyncConn:
-    AsyncRead + AsyncWrite + Connection + Send + Sync + Unpin + 'static
+pub trait AsyncConn:
+    AsyncRead + AsyncWrite + Connection + Send + Unpin + 'static
```

Comment:
the `Sync` bound appeared to be unused, and made implementing `AsyncConn` a bit annoying in some cases. It could be re-added if you prefer, at the cost of requiring a mutex for some implementations (like for Arti `DataStream`, which is `Send` but `!Sync`